src/sync.js

```diff
@@ -30,136 +30,201 @@ function decrypt(key, encrypted) {
   });
 }
 
+// An "id schema" used to validate Kinto IDs and generate new ones.
+const notesIdSchema = {
+  // FIXME: Maybe this should generate IDs?
+  generate() {
+    throw new Error("cannot generate IDs");
```

Comment:
Once the `encode`/`decode` issues are resolved below, I believe ESLint will start complaining about use of double quotes instead of single quotes here. Sorry.